js.go

```diff
@@ -1204,13 +1247,55 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 		consumer = info.Name
 		deliver = info.Config.DeliverSubject
 	}
-	sub.mu.Lock()
-	sub.jsi.stream = stream
-	sub.jsi.consumer = consumer
-	sub.jsi.durable = isDurable
-	sub.jsi.attached = attached
-	sub.jsi.deliver = deliver
-	sub.mu.Unlock()
+
+	// Recent servers are more tolerant for ephemerals not immediately being present so we can do sub here.
```

Comment:
Actually we can't delay the creation of the subscription to that point. The reason is that if user provides a durable name and say create 10 go routines that try to create the durable at the same time, all AddConsumer will return OK but the server will update the deliver subject with whatever latest call was. Out of the 10 durable subs, only 1 is going to receive this.
I have a test in the C client to test for that (originally I had also delayed creation of the sub here, but realized in code coverage analysis that the "already exists" condition was never reached and figured out what was happening, as described above).

Reply:
Will write that test and see if I can move things around where sub is prior to consumer create but still have the jsi context that is needed.

Reply:
ok changed.